RELEASE.md

```diff
@@ -0,0 +1,53 @@
+# Boxo Releases
```

Comment:
Move this under docs/ ?  Not sure the best move here, but I was thinking the FAQ.md should be there per https://github.com/ipfs/boxo/pull/260